### PR TITLE
f strings in core/series

### PR DESCRIPTION
**Pull request by donehower**
- [ ] references #29547
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -2944,7 +2944,7 @@ def _try_kind_sort(arr):
             sortedIdx[n:] = idx[good][argsorted]
             sortedIdx[:n] = idx[bad]
         else:
-            raise ValueError("invalid na_position: {!r}".format(na_position))
+            raise ValueError(f"invalid na_position: {na_position}")
```
Comment (WillAyd) [inline pandas/core/series.py]:
```suggestion
            raise ValueError(f"invalid na_position: {repr(na_position)}")
```

OK to replace the previous `!r` specifier with a `repr` call here

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good and I think CI failures are unrelated / have been fixed on master since this was created.

To be sure, can you locally on your `firstcommit` branch run:

```sh
git fetch upstream
git merge upstream/master
git push origin firstcommit
```

I think should get CI green for this. Flag me down at PyData if you have any questions

File: pandas/core/series.py [hunk 2]
```diff
@@ -104,9 +104,9 @@ def _coerce_method(converter):
     def wrapper(self):
         if len(self) == 1:
             return converter(self.iloc[0])
-        raise TypeError("cannot convert the series to {0}".format(str(converter)))
+        raise TypeError(f"cannot convert the series to {str(converter)}")
```
Comment (simonjayhawkins) [inline pandas/core/series.py]:
```suggestion
        raise TypeError(f"cannot convert the series to {converter}")
```
I think the str is redundant.

File: pandas/core/series.py [hunk 3]
```diff
@@ -274,8 +274,8 @@ def __init__(
                 try:
                     if len(index) != len(data):
                         raise ValueError(
-                            "Length of passed values is {val}, "
-                            "index implies {ind}".format(val=len(data), ind=len(index))
+                            f"Length of passed values is {len(data)}, "
+                            "index implies {len(index)}."
```
Comment (simonjayhawkins) [inline pandas/core/series.py]:
```suggestion
                            f"index implies {len(index)}."
```
needs an `f` here too.

File: pandas/core/series.py (hunk 1 above)
Comment (jreback) [reply pandas/core/series.py]: don't need repr on a boolean

**Comment by jreback** [comment]
pls merge master as well

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @donehower keep them coming

**Pull request merged**